firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java

```diff
@@ -620,14 +648,16 @@ public Void call() throws Exception {
   void writeNonFatalException(final Thread thread, final Throwable ex) {
     // Capture and close over the current time, so that we get the exact call time,
     // rather than the time at which the task executes.
-    final Date now = new Date();
+    final Date time = new Date();
 
     backgroundWorker.submit(
         new Runnable() {
           @Override
           public void run() {
             if (!isHandlingException()) {
-              doWriteNonFatal(now, thread, ex);
+              long timestampSeconds = time.getTime() / 1000;
+              reportingCoordinator.persistNonFatalEvent(ex, thread, timestampSeconds);
+              doWriteNonFatal(thread, ex, timestampSeconds);
```

Comment:
it's unfortunate the order changes between these two events

Reply:
Can you elaborate? I'm not sure what you mean.

Reply:
Talked about this offline and the reasoning for it, and we're happy with the implementation as-is.